### PR TITLE
chore: Add Stepbar and ArrowLink components

### PR DESCRIPTION
**Pull request by eric-zero-dev**
1. Added `$size-xlarge` for 32px of font size
2. Added Close icon and onClose props in Modal
3. Added Stepbar and ArrowLinkComponent

### CODE REVIEW

File: src/components/Icons/IconClose.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
Comment (colbr) [inline src/components/Icons/IconClose.tsx:1]: Can we move this Icon to the same folder as `Modal`? It's not used anywhere else (yet, at least), so best to keep it with `Modal`.

File: src/components/Input/Input.tsx [hunk 2]
```diff
@@ -16,10 +16,10 @@ interface InputProps {
...
 
   // these come from react-aria
-  labelProps: LabelHTMLAttributes<HTMLLabelElement>;
-  inputProps: InputHTMLAttributes<HTMLInputElement>;
-  descriptionProps: HTMLAttributes<HTMLElement>;
-  errorMessageProps: HTMLAttributes<HTMLElement>;
+  labelProps?: LabelHTMLAttributes<HTMLLabelElement>;
+  inputProps?: InputHTMLAttributes<HTMLInputElement>;
+  descriptionProps?: HTMLAttributes<HTMLDivElement>;
+  errorMessageProps?: HTMLAttributes<HTMLDivElement>;
```
Comment (colbr) [inline src/components/Input/Input.tsx:22]: Can we put this in a separate PR?

File: src/components/Link/ArrowLink/ArrowLink.module.scss [hunk 3]
```diff
@@ -0,0 +1,78 @@
+@use '../../../styles/animations';
```
Comment (colbr) [inline src/components/Link/ArrowLink/ArrowLink.module.scss:1]: I know this file was just pulled from `zNS-dapp`, but can you refactor to handle proper SCSS nesting and convert `px` to `rem`?

File: src/components/ProgressIndicators/StepBar/StepBar.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    <div className={classNames(styles.StepBar, className)}>
+      {steps.map((s: string, i: number) => {
+        const isVisible = step - 1 > i;
+        const cursor = isVisible ? "pointer" : "default";
+        return (
+          <div
+            key={i + s}
+            className={`${styles.Placeholder} ${isVisible ? styles.Show : ""}`}
```
Comment (colbr) [inline src/components/ProgressIndicators/StepBar/StepBar.tsx:44]: Use `classnames/bind`.

Also, is there any way we can completely hide hidden steps, rather than just making them `opacity: 0`? `opacity: 0` will make accessibility very difficult.

File: src/components/ProgressIndicators/StepBar/StepBar.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+          </div>
+        );
+      })}
+
+      <div
+        style={{
+          width,
+          transform: `translateX(${translate})`,
+        }}
+        className={`${styles.Bar} ${step > steps.length ? styles.Hide : ""}`}
```
Comment (colbr) [inline src/components/ProgressIndicators/StepBar/StepBar.tsx:65]: Use `classNames(...)`

File: src/components/ProgressIndicators/StepBar/StepBar.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  step: number;
+  steps: string[];
+  stepFormatter?: string;
+  onNavigate?: (i: number) => void;
+  className?: string;
+};
+
+export const StepBar: React.FC<StepBarProps> = ({ step, steps, stepFormatter, onNavigate, className }) => {
+  const translate = step >= steps.length ? `${(steps.length - 1) * 100}%` : `${(step - 1) * 100}%`;
+  const width = `${(1 / steps.length) * 100}%`;
```
Comment (colbr) [inline src/components/ProgressIndicators/StepBar/StepBar.tsx:18]: nit: might be cleaner to calculate the number here, and do the string concatenation in place.

File: src/components/ProgressIndicators/index.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+export * from "./StepBar";
```
Comment (colbr) [inline src/components/ProgressIndicators/index.ts:1]: Put `StepBar` in it's own folder in `components`, rather than nesting it in `ProgressIndicators`

File: src/components/Modal/index.tsx [hunk 8]
```diff
@@ -44,6 +47,11 @@ const Modal: FC<ModalProps> = ({ className, children, trigger, ...rest }) => {
       )}
       <Dialog.Portal>
         <Overlay>
+          {onClose && (
+            <Dialog.Close className={styles.CloseButton} onClick={onClose}>
+              <IconClose />
+            </Dialog.Close>
+          )}
```
Comment (colbr) [inline src/components/Modal/index.tsx:54]: Radix Dialog has an `onOpenChange` prop already - can we tie our `onClose` into that somehow?

File: src/components/ProgressIndicators/StepBar/StepBar.tsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    return `Step ${i + 1} of ${steps.length}: ${stepName}`;
+  };
+
+  return (
+    <div className={classNames(styles.StepBar, className)}>
+      {steps.map((s: string, i: number) => {
+        const isVisible = step - 1 > i;
+        const cursor = isVisible ? "pointer" : "default";
+        return (
+          <div
```
Comment (colbr) [inline src/components/ProgressIndicators/StepBar/StepBar.tsx:42]: These `div`s should actually be buttons for better accessibility. Use `useButton` from `react-aria` for this.

File: src/components/ProgressIndicators/StepBar/StepBar.tsx [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    if (step - 1 >= i) onNavigate?.(i);
+  };
+
+  const text = (stepName: string, i: number) => {
+    if (stepFormatter) {
+      return stepFormatter
+        .replace(/TOTAL_STEP_COUNT/g, String(steps.length))
+        .replace(/CURRENT_STEP_INDEX/g, String(i))
+        .replace(/CURRENT_STEP_NAME/g, stepName);
+    }
```
Comment (colbr) [inline src/components/ProgressIndicators/StepBar/StepBar.tsx:31]: Does `StepBar` text formatting ever change? If it does it's probably a problem in the design system - we shouldn't have to pass this in (at least, I can't think of anywhere we should).

File: src/styles/_typography.scss [hunk 11]
```diff
@@ -1,4 +1,5 @@
 // Font Size
+$size-xlarge: 2rem; // 32px
```
Comment (colbr) [inline src/styles/_typography.scss:2]: Thinking of deprecating these font sizes, as they aren't defined in the design system.

File: src/components/Link/ArrowLink/ArrowLink.tsx [hunk 12]
```diff
@@ -0,0 +1,71 @@
+//- React Imports
```
Comment (colbr) [inline src/components/Link/ArrowLink/ArrowLink.tsx:1]: Not your fault obviously, but this component is actually pretty weird. We should simplify this down a lot. If it's going to be a link it should probably have a similar/identical interface to HTML `a` elements.

**Review by colbr — CHANGES_REQUESTED** [review]
I recognise that this is just bought over from `zNS-dapp` and none of the requested changes are for your code (in fact, it's probably all for mine, lol). On reviewing these old components I think we actually need to rewrite them a bit to make them cleaner and more accessible.

**Comment by eric-zero-dev** [comment]
> I recognise that this is just bought over from `zNS-dapp` and none of the requested changes are for your code (in fact, it's probably all for mine, lol). On reviewing these old components I think we actually need to rewrite them a bit to make them cleaner and more accessible.

Yeah, I updated some codes of `StepBar` component. but not entire component

File: src/components/Input/Input.tsx (hunk 2 above)
Comment (eric-zero-dev) [reply src/components/Input/Input.tsx:22]: I had to update it here because it causes the ERROR in zOS app

File: src/components/Link/ArrowLink/ArrowLink.module.scss (hunk 3 above)
Comment (eric-zero-dev) [reply src/components/Link/ArrowLink/ArrowLink.module.scss:1]: Never checked this scss file

File: src/components/ProgressIndicators/StepBar/StepBar.tsx (hunk 5 above)
Comment (eric-zero-dev) [reply src/components/ProgressIndicators/StepBar/StepBar.tsx:65]: Same, never changed this part

File: src/components/ProgressIndicators/StepBar/StepBar.tsx (hunk 6 above)
Comment (eric-zero-dev) [reply src/components/ProgressIndicators/StepBar/StepBar.tsx:18]: same, never changed this part

File: src/components/Modal/index.tsx (hunk 8 above)
Comment (eric-zero-dev) [reply src/components/Modal/index.tsx:54]: I think onClose is more generic for modal. But its up to your approval

File: src/components/ProgressIndicators/StepBar/StepBar.tsx (hunk 9 above)
Comment (eric-zero-dev) [reply src/components/ProgressIndicators/StepBar/StepBar.tsx:42]: same, never changed this part

File: src/components/ProgressIndicators/StepBar/StepBar.tsx (hunk 10 above)
Comment (eric-zero-dev) [reply src/components/ProgressIndicators/StepBar/StepBar.tsx:31]: Please check buy domains design and old text presentation. It is different

File: src/styles/_typography.scss (hunk 11 above)
Comment (eric-zero-dev) [reply src/styles/_typography.scss:2]: But there is 32px font size in buy domain design

File: src/components/Link/ArrowLink/ArrowLink.tsx (hunk 12 above)
Comment (eric-zero-dev) [reply src/components/Link/ArrowLink/ArrowLink.tsx:1]: same, never changed this component

File: src/styles/_typography.scss (hunk 11 above)
Comment (colbr) [reply src/styles/_typography.scss:2]: I don't think Figma variables have been set up for font-sizes, so we don't get any benefit from storing them as SCSS variables.

**Comment by eric-zero-dev** [comment]
@colbr I have updated the stepbar. Can we merge it? Thanks

**Review by colbr — APPROVED** [review]
Approved, but we should refactor `ArrowLink`

File: src/components/Wizard/index.tsx [hunk 13]
```diff
@@ -11,7 +11,7 @@ import NFTDetails from "./Presets/NFTDetails";
 import "./Wizard.scss";
 
 interface HeaderProps {
-  header?: string;
+  header?: ReactNode;
```
Comment (colbr) [inline src/components/Wizard/index.tsx:14]: This should be `ReactNode | string`

File: src/components/Button/index.tsx [hunk 14]
```diff
@@ -7,7 +7,7 @@ import classNames from "classnames";
 import "./Button.scss";
 import "focus-visible";
 
-type ButtonProps = {
+type ButtonProps  = {
```
Comment (colbr) [inline src/components/Button/index.tsx:10]: Remove extra space.

**Review by colbr — CHANGES_REQUESTED** [review]

File: src/components/Wizard/index.tsx (hunk 13 above)
Comment (eric-zero-dev) [reply src/components/Wizard/index.tsx:14]: string is the part of ReactNode

File: src/components/ProgressIndicators/StepBar/StepBar.tsx (hunk 4 above)
Comment (eric-zero-dev) [reply src/components/ProgressIndicators/StepBar/StepBar.tsx]: I updated this part in previous commit.

File: src/components/Wizard/index.tsx (hunk 13 above)
Comment (eric-zero-dev) [reply src/components/Wizard/index.tsx:14]: ` type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;`
`type ReactChild = ReactElement | ReactText;`
`type ReactText = string | number;`

Hope you understand it.

File: src/components/Button/index.tsx (hunk 14 above)
Comment (eric-zero-dev) [reply src/components/Button/index.tsx:10]: not sure why it is added, but is there husky integration for this repo?

File: src/components/Wizard/index.tsx (hunk 13 above)
Comment (eric-zero-dev) [reply src/components/Wizard/index.tsx:14]: I just remember that I was going to add close icon in `header`, but not added it yet. I think it should be handle in Wizard

File: src/components/ProgressIndicators/StepBar/StepBar.tsx (hunk 4 above)
Comment (eric-zero-dev) [reply src/components/ProgressIndicators/StepBar/StepBar.tsx]: I mean it is refactored. Could you please check?

File: src/components/Button/index.tsx (hunk 14 above)
Comment (eric-zero-dev) [reply src/components/Button/index.tsx]: I just checked the button/index.tsx. I was really bad.

https://github.com/zer0-os/zUI/pull/1/files#diff-2a0d77decd98a62a279ac9fb139cad79d6d7a20f4c724de3dc811911bc5dd9b1L1-R1

File: src/components/Wizard/index.tsx (hunk 13 above)
Comment (eric-zero-dev) [reply src/components/Wizard/index.tsx:14]: I can see there are some places you used ` ReactNode | string`, those should be `ReactNode`

Comment (colbr) [reply src/components/Wizard/index.tsx:14]: Oh I see - makes sense I didn't know that! Thanks 🚀

**Review by colbr — APPROVED** [review]
